consensus/ethash/consensus.go

```diff
@@ -273,19 +273,16 @@ func (ethash *Ethash) verifyHeader(chain consensus.ChainReader, header, parent *
 		return fmt.Errorf("invalid gasUsed: have %d, gasLimit %d", header.GasUsed, header.GasLimit)
 	}
 
-	// TODO: UNCOMMENT THIS CHECK WHEN WE UNDERSTAND OUR GAS LIMIT REQUIREMENTS
-
 	// Verify that the gas limit remains within allowed bounds
-	//diff := int64(parent.GasLimit) - int64(header.GasLimit)
-	//if diff < 0 {
-	//	diff *= -1
-	//}
-
-	//limit := parent.GasLimit / params.GasLimitBoundDivisor
-	//if uint64(diff) >= limit || header.GasLimit < params.MinGasLimit {
-	//	return fmt.Errorf("invalid gas limit: have %d, want %d += %d", header.GasLimit, parent.GasLimit, limit)
-	//}
+	diff := int64(parent.GasLimit) - int64(header.GasLimit)
+	if diff < 0 {
+		diff *= -1
+	}
+	limit := parent.GasLimit / params.GasLimitBoundDivisor
 
+	if uint64(diff) >= limit || header.GasLimit < params.MinGasLimit {
```

Comment:
This errors out because `header.GasLimit < params.MinGasLimit` when called [here](https://github.com/ethereum-optimism/go-ethereum/blob/master/core/blockchain_test.go#L877). It happens during the 3rd block in the chain.

the `MinGasLimit = 4m`, whereas the gas limit used in the provided header is 3.4m